runners/flink/src/main/java/org/apache/beam/runners/flink/translation/functions/FlinkDefaultExecutableStageContext.java

```diff
@@ -64,19 +71,60 @@ public void close() throws Exception {
     jobBundleFactory.close();
   }
 
-  enum ReferenceCountingFactory implements Factory {
-    REFERENCE_COUNTING;
+  private static class JobFactoryState {
+    private final AtomicInteger counter = new AtomicInteger(0);
```

Comment:
Why is this AtomicInteger? AFAIK it is only access in a synchronized method.

Reply:
Yeah, this doesn't need to be AtomicInteger. Left over from an earlier version that was not synchronized.